semgrep-core/src/engine/Match_rules.ml

```diff
@@ -986,8 +986,7 @@ let check_rule r hook default_config pformula equivs file_and_more =
 
 let check hook default_config rules equivs file_and_more =
   let { FM.file; lazy_ast_and_errors; _ } = file_and_more in
-  logger#info "checking %s with %d rules" file (List.length rules);
-  if rules = [] then logger#error "empty rules";
```

Comment:
not sure why you removed the empty rules logger.

Reply:
Despite being an "error" this happens very often either because we get a taint rule, or because the rule is not relevant for the file. If we don't consider it an error, then it's somewhat redundant with the previous logger. But I can add it back, I would just suggest that then we change the code so that `check` is not called with an empty list of rules.

Reply:
We would have to change this in Run_rules.ml
```ocaml
let check hook default_config rules equivs file_and_more =
  let search_rules, taint_rules, skipped_rules =
    filter_and_partition_rules rules file_and_more
  in
  let res_search =
    Match_rules.check hook default_config search_rules equivs file_and_more
  in
```

Reply:
I'll merge, but if you want us to do something about this just let me know.